<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>R/simcross User Guide</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: #990073
   }

   pre .number {
     color: #099;
   }

   pre .comment {
     color: #998;
     font-style: italic
   }

   pre .keyword {
     color: #900;
     font-weight: bold
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: #d14;
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>



<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
</style>



</head>

<body>
<!--
%\VignetteEngine{knitr::knitr}
%\VignetteIndexEntry{R/simcross User Guide}
-->

<h1>R/simcross User Guide</h1>

<p><a href="https://github.com/kbroman/simcross">R/simcross</a> is an
<a href="http://www.r-project.org">R</a> package to simulate genotypes from an
experimental cross. The aim is flexibility rather than speed.</p>

<p>Meiosis is simulated following the Stahl model (Copenhaver et
al. 2002) with the interference parameter being an integer. In this
model, chiasmata locations are the superposition of two processes: a
proportion <em>p</em> come from a process exhibiting no interference (that
is, a Poisson process) and the remainder (proportion 1 &ndash; <em>p</em>)
come from a process following the chi-square model (Foss et al. 1993;
Zhao et al. 1995). Thus, with <em>p</em>=0, the model reduces to the
chi-square model.  The chi-square model has a single parameter, <em>m</em>,
which is a non-negative integer and controls the strength of
interference. <em>m</em>=0 corresponds to no interference.  Broman et
al. (2002) estimated <em>m</em>=10 for the level of interference in the mouse
(derived assuming <em>p</em>=0).  The chi-square model is a special case of
the gamma model (McPeek and Speed 1995) which has a positive parameter
<em>&nu;</em>; with <em>&nu;</em> = <em>m</em> + 1.</p>

<p>In many organisms, there is always at least one chiasma on the
four-strand bundle at meiosis. Simulations with
<a href="https://github.com/kbroman/simcross">R/simcross</a> may be performed
with this assumption; the same model is used, but with rejection
sampling to get results conditional on there being at least one
chiasma. Note that this isn&#39;t an assumption of an obligate
<em>crossover</em>; if there is exactly one chiasma on the four-strand
bundle, then a random meiotic product will have 0 or 1 crossover, with
probability &frac12; each. Also note that, with the obligate chiasma
assumption, chromosomes must be greater than 50 cM. And note that</p>

<h2>Basic functions</h2>

<p>There are two basic functions for simulating a cross:
<code>create_parent</code> and <code>cross</code>.</p>

<p><code>create_parent</code> parent is for generating a parent object, either an
inbred individual or the F<sub>1</sub> offspring of two inbred
individuals. It takes two arguments: the length of the chromosome in
cM, and the allele or pair of alleles.</p>

<p>Here&#39;s how to generate two inbred individuals, one from each of strain
1 and strain 2, and an F<sub>1</sub> individual, with a 100 cM
chromosome.</p>

<pre><code class="r">library(simcross)
p1 &lt;- create_parent(100, 1)
p2 &lt;- create_parent(100, 2)
f1 &lt;- create_parent(100, 1:2)
</code></pre>

<p>The <code>cross</code> function is used to generate one offspring from the cross
between two individuals.  The input is a pair of individuals (e.g., as
produced by <code>create_parent</code>), the interference parameter <code>m</code> (default
<code>m=10</code>), the parameter <code>p</code> for the Stahl model (default <code>p=0</code>), <code>xchr</code>
to indicate whether the chromosome being simulated is the X chromosome
(<code>FALSE</code>, the default, to simulate an autosome), and <code>male</code> to
indicate whether the offspring is to be male (which only matters if
<code>xchr=TRUE</code>). Further, one may use <code>obligate_chiasma=TRUE</code> (default
<code>FALSE</code>) to require at least one chiasma on the 4-strand bundle).</p>

<p>Here&#39;s an example, to generate an F<sub>2</sub> individual.  (The
outer parentheses cause the result to be printed.)</p>

<pre><code class="r">(f2 &lt;- cross(f1, f1))
</code></pre>

<pre><code>## $mat
## $mat$alleles
## [1] 2
## 
## $mat$locations
## [1] 100
## 
## 
## $pat
## $pat$alleles
## [1] 1 2 1
## 
## $pat$locations
## [1]  19.41  51.34 100.00
</code></pre>

<p>The output is list with two components: the maternal and paternal
chromosomes. Each chromsome is a list with the allele in a set
of intervals, and the locations of the right endpoints of the
intervals.</p>

<p>In the example above, the maternal chromosome is a non-recombinant <code>2</code>
chromosome. The paternal chromosome has two crossovers, with the
the <code>1</code> allele up to 19.4 cM, the <code>2</code> allele
in the interval 19.4 &ndash;
51.3, and the <code>1</code> allele
for the remainder of the chromosome.</p>

<p>Behind the scenes, there are two additional
functions, <code>sim_crossovers</code>, for simulating crossover locations on a
chromosome, and <code>sim_meiosis</code>, for simulating a meiotic product from
an individual, but in general the user need not bother with these.
The <code>cross</code> function calls <code>sim_meiosis</code> twice
(once for each parent) and then combines the results into a single
individual. <code>sim_meiosis</code> calls <code>sim_crossovers</code> to generate the
meiotic product.</p>

<h2>Generating pedigrees</h2>

<p>While one could simulate any experimental cross from a series of calls
to <code>create_parent</code> and <code>cross</code>, it is generally more efficient to first develop
a table that describes pedigree for the cross, and then simulate from
the pedigree with the function <code>sim_from_pedigree</code> (see the next
section).</p>

<h3>Example AIL pedigree matrix</h3>

<p>To define a pedigree, we use a numeric matrix with four columns:
individual ID, mom, dad, and sex (coded as 0=female, 1=male). The
<a href="https://github.com/kbroman/simcross">R/simcross package</a> includes a
sample pedigree for advanced intercross lines, <code>AILped</code>, taken from
the <a href="http://cran.r-project.org/web/packages/QTLRel/">QTLRel package</a>.
Here is the top of that dataset:</p>

<pre><code class="r">head(AILped, n=10)
</code></pre>

<pre><code>##       id mom dad sex generation
##  [1,]  1   0   0   1          0
##  [2,]  2   0   0   0          0
##  [3,]  3   2   1   1          1
##  [4,]  4   2   1   0          1
##  [5,] 15   4   3   0          2
##  [6,] 16   4   3   0          2
##  [7,] 17   4   3   0          2
##  [8,] 14   4   3   1          2
##  [9,] 11   4   3   1          2
## [10,] 12   4   3   1          2
</code></pre>

<h3>Checking a pedigree matrix</h3>

<p>The function <code>check_pedigree</code> can be used to check that a pedigree
matrix conforms to R/simcross&#39;s requirements: founders have
\code{mom == dad == 0}, all other individuals have both parents
present in the pedigree, and parents always preceded any of their
children. The <code>check_pedigree</code> function returns <code>TRUE</code> if the pedigree
matrix is okay; otherwise, it throws an error.</p>

<pre><code class="r">check_pedigree(AILped)
</code></pre>

<pre><code>## [1] TRUE
</code></pre>

<p>R/simcross includes a set of functions for generating pedigree
matrices for different cross designs: <code>sim_ril_pedigree</code>,
<code>sim_4way_pedigree</code>, <code>sim_ail_pedigree</code>, and <code>sim_do_pedigree</code>.</p>

<h3>RIL pedigree</h3>

<p><code>sim_ril_pedigree</code> generates a pedigree matrix for a single
recombinant inbred line. The arguments are <code>ngen</code> (number of
generations of inbreeding), <code>selfing</code> (default <code>FALSE</code>, for sibling
mating), <code>parents</code> (a vector of integers for the parents; the length
must be 2, 4, or 8, and corresponds to the number of founder lines),
and <code>firstind</code>, the ID number to attach to the first individual
following the parents (so that the pedigree matrices for multiple RIL
may be <code>rbind</code>-ed together). Not that there&#39;s no real <em>simulation</em>
here; the result is entirely deterministic.)</p>

<pre><code class="r">(ril &lt;- sim_ril_pedigree(ngen=4, parents=1:4))
</code></pre>

<pre><code>##       id mom dad sex gen
##  [1,]  1   0   0   0   0
##  [2,]  2   0   0   1   0
##  [3,]  3   0   0   0   0
##  [4,]  4   0   0   1   0
##  [5,]  5   1   2   0   1
##  [6,]  6   3   4   1   1
##  [7,]  7   5   6   0   2
##  [8,]  8   5   6   1   2
##  [9,]  9   7   8   0   3
## [10,] 10   7   8   1   3
## [11,] 11   9  10   0   4
## [12,] 12   9  10   1   4
## [13,] 13  11  12   0   5
## [14,] 14  11  12   1   5
## [15,] 15  13  14   0   6
## [16,] 16  13  14   1   6
</code></pre>

<h3>4-way cross pedigree</h3>

<p><code>sim_4way_pedigree</code> generates a pedigree matrix for an intercross
among four inbred lines. The arguments are <code>ngen</code> (which must be 1 or
2) and <code>nsibs</code>. We start with four inbred individuals, and cross them
in two pairs to generate a pair of heterozygous individuals. If
<code>ngen==1</code>, we then just generate a set of <code>sum(nsibs)</code> F<sub>1</sub>
offspring. If <code>ngen==2</code>, we generate <code>length(nsibs)</code> pairs of
F<sub>1</sub>s and intercross them to generate a set of F<sub>2</sub>
sibships; in this case, the input vector <code>nsibs</code> determines the sizes
of the sibships.</p>

<p>The following generates two F<sub>2</sub> sibships with 3
offspring in each.</p>

<pre><code class="r">(fourway &lt;- sim_4way_pedigree(2, c(3, 3)))
</code></pre>

<pre><code>##       id mom dad sex gen
##  [1,]  1   0   0   0   0
##  [2,]  2   0   0   1   0
##  [3,]  3   0   0   0   0
##  [4,]  4   0   0   1   0
##  [5,]  5   1   2   0   1
##  [6,]  6   3   4   1   1
##  [7,]  7   5   6   0   2
##  [8,]  8   5   6   1   2
##  [9,]  9   5   6   0   2
## [10,] 10   5   6   1   2
## [11,] 11   7   8   0   3
## [12,] 12   7   8   1   3
## [13,] 13   7   8   0   3
## [14,] 14   9  10   1   3
## [15,] 15   9  10   0   3
## [16,] 16   9  10   1   3
</code></pre>

<h3>AIL pedigree</h3>

<p><code>sim_ail_pedigree</code> generates a pedigree matrix for 2-way advanced
intercross lines.</p>

<h3>Diversity Outcross pedigree</h3>

<h2>Simulating from a pedigree</h2>

<p>The function <code>sim_from_pedigree</code>.</p>

<h2>Generating marker data</h2>

<p>The function <code>convert2geno</code> and <code>get_geno</code>.</p>

<h2>References</h2>

<p>Broman KW, Rowe LB, Churchill GA, Paigen K (2002)
<a href="http://www.genetics.org/content/160/3/1123.short">Crossover interference in the mouse</a>.
Genetics 160:1123-1131</p>

<p>Copenhaver GP, Housworth EA, Stahl FW (2002)
<a href="http://www.genetics.org/content/160/4/1631.short">Crossover interference in arabidopsis</a>.
Genetics 160:1631-1639</p>

<p>Foss E, Lande R, Stahl FW, Steinberg CM (1993)
<a href="http://www.genetics.org/content/133/3/681.short">Chiasma interference as a function of genetic distance</a>.
Genetics 133:681-691</p>

<p>McPeek MS, Speed TP (1995)
<a href="http://www.genetics.org/content/139/2/1031.short">Modeling interference in genetic recombination</a>.
Genetics 139:1031-1044</p>

<p>Svenson KL, Gatti DM, Valdar W, Welsh CE, Cheng R, Chesler EJ, Palmer
AA, McMillan L, Churchill GA (2012)
<a href="http://www.genetics.org/content/190/2/437.short">High-resolution genetic mapping using the mouse Diversity Outbred population.</a>
Genetics 190:437-447</p>

<p>Zhao H, Speed TP, McPeek MS (1995)
<a href="http://www.genetics.org/content/139/2/1045.short">Statistical analysis of crossover interference using the chi-square model</a>.
Genetics 139:1045-1056</p>

<!-- the following to make it look nicer -->

<p><link href="http://kbroman.org/qtlcharts/assets/vignettes/vignette.css" rel="stylesheet"></link></p>

</body>

</html>
